src/init.cpp

```diff
@@ -1181,6 +1181,20 @@ bool AppInit2(ThreadHandlerPtr threads)
         return false;
     }
 
+    // This is for the second part of the reindex. We need the set to ensure the order is correct.
+    std::set<std::pair<fs::path, uintmax_t>> block_data_files_to_load;
```

Comment:
Are you sure about using a set here? This will override the user's `-loadblock` order.

Reply:
That is a good point. Using a sort is required when dealing with boost filesystem iterators, because the order of the files in the iterator is not guaranteed. This is important for the -reindex mode. I think what I will do is change this to a vector, and then use a sort bool parameter that will cause the vector to be sorted or leave it alone, because we want the -reindex to be sorted, but the -loadblock to respect the parameter order.